src/Microsoft.DotNet.Cli.Utils/CommandResolution/DepsJsonCommandResolver.cs

```diff
@@ -0,0 +1,261 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Microsoft.DotNet.ProjectModel;
+using Microsoft.DotNet.ProjectModel.Graph;
+using Microsoft.DotNet.ProjectModel.Compilation;
+using Microsoft.Extensions.DependencyModel;
+using Microsoft.Extensions.PlatformAbstractions;
+using Microsoft.DotNet.ProjectModel.Resolution;
+using NuGet.Frameworks;
+using NuGet.Packaging;
+using NuGet.ProjectModel;
+
+using LockFile = Microsoft.DotNet.ProjectModel.Graph.LockFile;
+using FileFormatException = Microsoft.DotNet.ProjectModel.FileFormatException;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    public class DepsJsonCommandResolver : ICommandResolver
+    {
+        private static readonly string[] s_extensionPreferenceOrder = new [] 
+        { 
+            "",
+            ".exe",
+            ".dll"
+        };
+
+        private string _nugetPackageRoot;
+        private Muxer _muxer;
+
+        public DepsJsonCommandResolver(string nugetPackageRoot)
+        {
+            _muxer = new Muxer();
+            _nugetPackageRoot = nugetPackageRoot;
+        }
+
+        public DepsJsonCommandResolver(Muxer muxer, string nugetPackageRoot)
+        {
+            _muxer = muxer;
+            _nugetPackageRoot = nugetPackageRoot;
+        }
+
+        public CommandSpec Resolve(CommandResolverArguments commandResolverArguments)
+        {
+            if (commandResolverArguments.CommandName == null
+                || commandResolverArguments.DepsJsonFile == null)
+            {
+                return null;
+            }
+            
+            return ResolveFromDepsJsonFile(
+                commandResolverArguments.CommandName, 
+                commandResolverArguments.CommandArguments.OrEmptyIfNull(),
+                commandResolverArguments.DepsJsonFile);
+        }
+
+        private CommandSpec ResolveFromDepsJsonFile(
+            string commandName, 
+            IEnumerable<string> commandArgs, 
+            string depsJsonFile)
+        {
+            var dependencyContext = LoadDependencyContextFromFile(depsJsonFile);
```

Comment:
Ideally once per process invocation to launch fsc, I don't think we should optimize this for perf just yet given it should be low impact